web/src/main/java/org/akaza/openclinica/web/filter/OpenClinicaUsernamePasswordAuthenticationFilter.java

```diff
@@ -126,6 +119,24 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
             if (userAccountBean == null) {
                 throw new BadCredentialsException("Bad Credentials");
             }
+            /**
+             *  OC-12286
+             */
+            // check the user in the current session
+            HttpSession session  = null;
+            UserAccountBean currentSessionUser = null;
+            currentSessionUser = (UserAccountBean) request.getSession().getAttribute(SecureController.USER_BEAN_NAME);
+            
+            if(currentSessionUser !=null && !(currentSessionUser.equals(eb))) {
+            	request.getSession().invalidate();
+            	
+            }            
+        	// Place the last username attempted into HttpSession for views
+        	session = request.getSession(false);                    
+
+            if (session != null || getAllowSessionCreation()) {
+                request.getSession().setAttribute(SPRING_SECURITY_LAST_USERNAME_KEY, TextEscapeUtils.escapeEntities(username));
+            }
```

Comment:
please revert this change and implement the fix where userB would be prevented from logging in if userA already has an active session

Reply:
done